Mathlib/Topology/MetricSpace/MetricSeparated.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright (c) 2021 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+
+! This file was ported from Lean 3 source module topology.metric_space.metric_separated
+! leanprover-community/mathlib commit 57ac39bd365c2f80589a700f9fbb664d3a1a30c2
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Topology.MetricSpace.EMetricSpace
+
+/-!
+# Metric separated pairs of sets
+
+In this file we define the predicate `IsMetricSeparated`. We say that two sets in an (extended)
+metric space are *metric separated* if the (extended) distance between `x ∈ s` and `y ∈ t` is
+bounded from below by a positive constant.
+
+This notion is useful, e.g., to define metric outer measures.
+-/
+
+
+open EMetric Set
+
+noncomputable section
+
+/-- Two sets in an (extended) metric space are called *metric separated* if the (extended) distance
+between `x ∈ s` and `y ∈ t` is bounded from below by a positive constant. -/
+def IsMetricSeparated {X : Type _} [EMetricSpace X] (s t : Set X) :=
+  ∃ (r : _)(_ : r ≠ 0), ∀ x ∈ s, ∀ y ∈ t, r ≤ edist x y
```

Comment:
I would migrate to `∃ r, r ≠ 0 ∧ _` because we're going to do it anyway and it rarely causes compile errors down the path. Another option is to leave a porting note.